frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -287,6 +292,7 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
             Toast.error(messages["dataset.upload_failed"]);
             this.setState({
               isUploading: false,
+              isFinishing: true,
```

Comment:
```suggestion
              isFinishing: false,
```

probably? otherwise, the user might think that something will change if they wait. also `isFinishing` will already be true in that case, so it's a noop currently.

Reply:
Absolutely right. Didn’t show during testing since with isUploading being false, none of the uploading UI is rendered anyway. But this is certainly the cleaner state